exercises/ocr-numbers/example.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::HashMap;
+use std::str::Lines;
+
+fn lines_valid(l: &mut Lines) -> bool {
+    l.clone().count() % 4 == 0 && l.clone().all(|c| c.chars().count() % 3 == 0)
+}
+
+pub fn convert(input: &str) -> Result<String, ()> {
+    if !lines_valid(&mut input.lines()) {
+        return Err(());
+    }
+
+    let y = input.lines().collect::<Vec<_>>();
+
+    let mut converted_lines = Vec::new();
+
+    for char_lines in y.chunks(4) {
+        let mut unparsed_characters = HashMap::new();
+
+        for line in char_lines {
+            let line_chars = line.chars().clone().collect::<Vec<_>>();
+
+            for (char_number, char_chunk) in line_chars.chunks(3).enumerate() {
+                let char_chars = unparsed_characters.entry(char_number).or_insert(vec![]);
+                for c in char_chunk {
+                    char_chars.push(c.clone());
+                }
+            }
+        }
+
+        let mut parsed_characters = String::new();
+
+        for k in 0..unparsed_characters.len() {
```

Comment:
HashMap may not be the best choice, but I wanted to use `Entry`, which I can't use with a vec! (not that I can see, at least.)

I'll probably switch this to BTree, since that should solve the problem of ordering.